File: weekly_397th/B.cpp
class Solution {
public:
    int maximumEnergy(vector<int>& energy, int k) {
        int n = energy.size();
        int ans = INT_MIN;
        for (int i = n - k; i < n; i++) {
            int cur = 0;
            for(int j = i; j >= 0; j -= k) {
                cur += energy[j];
                ans = max(ans, cur);
            }
        }
        return ans;
    }
};

File: weekly_397th/B.go
func maximumEnergy(energy []int, k int) int {
    ans := math.MinInt
    n := len(energy)
    for i := n - k; i < n; i++ {
        cur := 0
        for j := i; j >= 0; j -= k {
            cur += energy[j]
            ans = int(math.Max(float64(ans), float64(cur)))
        }
    }
    return ans
}
